shared/src/main/java/uk/gov/di/authentication/shared/services/TokenService.java

```diff
@@ -300,6 +306,11 @@ private SignedJWT generateIDToken(
             idTokenClaims.setClaim("vot", vot);
         }
         idTokenClaims.setClaim("vtm", trustMarkUri.toString());
+        if (requiresResourceId && configService.isResourceIdScopeEnabled()) {
+            idTokenClaims.setClaim(
+                    CustomScopeValue.RESOURCE_ID.getValue(), subject.getValue().split(":")[4]);
+        }
```

Comment:
I think we should do this in userinfo by checking the scopes in the access token?

Reply:
I checked with them in the initial discussion because my preference was user-info too, and it needs to be in the identity token for the software they're using